workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -1,98 +1,58 @@
-            "The WorkerConnector behaviour was not found on the worker prefab";
-
-        private const int GuiPadding = 10;
-        private const float VerticalPositionRatio = 0.4f;
 
+        [SerializeField] private GameObject connectionPanel;
+        [SerializeField] private InputField ipAddressInput;
+        [SerializeField] private Button connectButton;
+        [SerializeField] private Text errorMessage;
         [SerializeField] private GameObject clientWorkerPrefab;
-        [SerializeField] private Font font;
-        [SerializeField] private float screenWidthFontRatio = 20;
-
-        private string ipAddressText;
-        private string errorMessage;
-        private bool isConnecting;
 
         private GameObject worker;
 
-        public void Awake()
-        {
-            ipAddressText = PlayerPrefs.GetString(HostIpPlayerPrefsKey);
-        }
+        private string IpAddress => ipAddressInput != null ? ipAddressInput.text : null;
 
-        public void OnGUI()
+        public void Awake()
         {
-            GUI.enabled = !isConnecting;
-            using (new GUILayout.AreaScope(new Rect(
-                GuiPadding,
-                Screen.height * VerticalPositionRatio,
-                Screen.width - GuiPadding * 2,
-                Screen.height - GuiPadding * 2)))
-            {
-                using (new ResizedGui(font, screenWidthFontRatio,
-                    GUI.skin.label,
-                    GUI.skin.textField,
-                    GUI.skin.button,
-                    GUI.skin.textArea
-                ))
-                {
-                    ResizedGui.Label("Enter IP address:");
-                    ipAddressText = ResizedGui.TextField(ipAddressText);
-
-                    if (ResizedGui.Button("Connect") && Application.isPlaying)
-                    {
-                        TryConnect();
-                    }
-
-                    if (!string.IsNullOrEmpty(errorMessage))
-                    {
-                        ResizedGui.TextArea($"Error: {errorMessage}");
-                    }
-                }
-            }
+            ipAddressInput.text = PlayerPrefs.GetString(HostIpPlayerPrefsKey);
+            connectButton.onClick.AddListener(TryConnect);
         }
 
         private void TryConnect()
         {
-            errorMessage = string.Empty;
-            isConnecting = true;
             worker = Instantiate(clientWorkerPrefab);
             var workerConnector = worker.GetComponent<IMobileConnectionController>();
 
             if (workerConnector == null)
             {
-                isConnecting  = false;
                 UnityObjectDestroyer.Destroy(worker);
-                errorMessage = MissingWorkerConnectorMessage;
-                throw new MissingComponentException(MissingWorkerConnectorMessage);
+                throw new MissingComponentException("The WorkerConnector behaviour was not found on the worker prefab");
```

Comment:
also set the error message here. that was nice

Reply:
Added back